tsdb/index/index.go

```diff
@@ -749,18 +782,21 @@ type Range struct {
 // for all postings lists.
 func (r *Reader) PostingsRanges() (map[labels.Label]Range, error) {
```

Comment:
As discussed offline, thanks for making this work. AFAIK it's only Thanos that uses this. As discussed about removing it: I would vote for keeping it as otherwise we need to mmap index twice to get mmaped bytes as it's private in reader ):

Reply:
Mmaps are cheap compared to what you're then doing with them.

Reply:
It did however get in my way, as a function with no uses, no unittests, but I still kinda have to keep working. This is why I really really don't like this sort of thing.